<!DOCTYPE html>
<html lang="en ">
<!-- Head Start -->
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="google-site-verification" content="xBT4GhYoi5qRD5tr338pgPM5OWHHIDR6mNg1a3euekI" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="keywords"  content="undefined,">
    <meta name="description" content="-">
    <meta property="og:locale" content="en">
    <meta property="og:type" content="article">
    <meta property="og:title" content="">
    <meta property="og:description" content="-">
    <meta property="og:url" content="http://example.com/2021/07/17/SNPGWASwyb/">
    <meta property="og:site_name" content="">
    <link rel="shortcut icon" href="/img/avatar/my-avatar.jpg">
    <title></title>
    <link rel="canonical" href="http://example.com/2021/07/17/SNPGWASwyb/">
    
<link rel="stylesheet" href="/css/font-awesome.min.css">

    
<link rel="stylesheet" href="/css/bootstrap.min.css">

    
<link rel="stylesheet" href="/css/omega.css">

    
<link rel="stylesheet" href="/css/highlight.css">

    
<link rel="stylesheet" href="/css/gitalk.min.css">

    
<link rel="stylesheet" href="/css/kkpager_orange.min.css">


    
<script src="/js/jquery.min.js"></script>

    
<script src="/js/bootstrap.min.js"></script>

    
<script src="/js/gitalk.min.js"></script>

    
<script src="/js/kkpager.min.js"></script>

    
<script src="/js/button.min.js"></script>

    
<script src="/js/fastclick.min.js"></script>

    
<script src="/js/omega.js"></script>

<meta name="generator" content="Hexo 5.4.0"></head>

<!-- Head End -->
<body ontouchstart="">
    <!-- Header Start -->
    
        
	<style>
		.post-header-no-banner {
			margin-top: 60px;
			margin-bottom: 20px;
			text-align: center;
			color: #333333;
		}
		.post-header-no-banner .meta {
			font-family: 'Lora', 'Times New Roman', serif;
			font-style: italic;
			font-weight: 300;
			font-size: 18px;
		}
	</style>
	<div class="post-header-no-banner">
		<div class="post-heading">
			<h1></h1>
			<h2 class="subheading"></h2>
			<div class="tags">
				
			</div>
			<span class="meta">
				Posted by John Doe on
				2021-07-17
			</span>
		</div>
	</div>

     
    <!-- Header End -->

    <!-- Nav Start -->
    <!-- Navigation -->
<nav class="navbar navbar-default navbar-custom navbar-fixed-top">
  <div class="container-fluid">
    <!-- Brand and toggle get grouped for better mobile display -->
    <div class="navbar-header page-scroll">
      <button type="button" class="navbar-toggle">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Hexo</a>
    </div>

    <div id="blog_navbar">
      <div class="navbar-collapse">
        <ul class="nav navbar-nav navbar-right">
          <li>
            <a href="#" id="search" ><img style="width: 16px;height: 16px;margin: -2px 5px 0px 0;" src="/icons/search.svg " alt="">search </a>
          </li>
          <li>
            <a href="/">home </a>
          </li>
           

          
        </ul>
      </div>
    </div>
  </div>
</nav>

<!-- Search Modal -->
<div class="modal search-modal">
  <div class="search-container">
    <div class="search-container-top">
      <img src="/icons/close.svg" alt="">
    </div>
    <input class="search-container-input" placeholder="placeholder " type="text">
    <div class="search-container-content"></div>
  </div>
</div>

<script>
  $('.search-container').css({ "height": window.innerHeight, "overflow": 'hidden' })
  $('.search-container-content').css({ "height": window.innerHeight - 70, "overflow": 'scroll' })
  var allArticles = [];
  var xReq = new XMLHttpRequest();
  xReq.open('GET', '/search.json');
  xReq.send();
  xReq.onreadystatechange = function(){
    if(xReq.readyState === 4 && xReq.status === 200){
      allArticles = JSON.parse(xReq.responseText);
    }
  }

  $('#search').on('click', function() {
    document.body.style['overflow-y'] = 'hidden'
    $('.search-modal').addClass('modal-show');
  });
  
  $('.search-modal,.search-container-top img').on('click', function() {
    document.body.style['overflow-y'] = 'scroll';
    $('.search-modal').removeClass('modal-show');
  });
  
  $('.search-container').on('click', function(e) {
    e.stopPropagation();
  });
  
  $('.search-container-input').bind('input propertychange', function(e) {
    var searchValue = e.currentTarget.value;
    var correctArticles = allArticles.filter(article => article.content.toLowerCase().indexOf(searchValue.toLowerCase()) > -1 || article.title.toLowerCase().indexOf(searchValue.toLowerCase()) > -1)
    var articleListHtml = '';
    for (let index = 0; index < correctArticles.length; index++) {
      var article = correctArticles[index];
      article.description = article.content.replace(/<[^>]+>/g,"");
      var title = article.title, description = '';
      var searchReg = RegExp(searchValue, 'i');
      var searchRegGlobal = RegExp(searchValue, 'ig');
      if (article.title.match(searchReg)) {
        title = article.title.replaceAll(searchRegGlobal, `<span class="search-value-active">${searchValue}</span>`);
      }
      var contentMatch = article.description.match(searchReg);
      if (contentMatch) {
        description = `...${article.description.substr(contentMatch['index'], 210)}...`.replaceAll(searchRegGlobal, `<span class="search-value-active">${searchValue}</span>`);
      } else {
        description = article.description.substr(0, 200);
      }

      articleListHtml += `<div class="search-container-content_item">
        <a href="${article.url}" class="search-container-content_item-title">${title}</a>
        <div class="search-container-content_item-description">${description}</div>
      </div>`
    }

    $('.search-container-content').html(articleListHtml);
  });
  
</script>
<script>
  var $body = document.body;
  var $toggle = document.querySelector(".navbar-toggle");
  var $navbar = document.querySelector("#blog_navbar");
  var $collapse = document.querySelector(".navbar-collapse");

  $toggle.addEventListener("click", handleMagic);
  function handleMagic(e) {
    if ($navbar.className.indexOf("in") > 0) {
      $navbar.className = " ";
      // wait until animation end.
      setTimeout(function () {
        // prevent frequently toggle
        if ($navbar.className.indexOf("in") < 0) {
          $collapse.style.height = "0px";
        }
      }, 400);
    } else {
      $collapse.style.height = "auto";
      $navbar.className += " in";
    }
  }
</script>

    <!-- Nav End -->

    <!-- Main Content Start -->
    <div class="main-content content-post ">
        
<!-- Post Content Start -->
<article>
    <div class="container" style="overflow: hidden">
        <div class="row">

            <!-- Post Container -->
            <div class="
                col-lg-8 col-lg-offset-2
                col-md-10 col-md-offset-1
                post-container post-markdown-body
            ">
                <p>[TOC]</p>
<p><strong>by wuyubing 2021/07/12</strong></p>
<h1 id="常规SNP-GWAS分析"><a href="#常规SNP-GWAS分析" class="headerlink" title="常规SNP GWAS分析"></a>常规SNP GWAS分析</h1><p>GWAS脚本复演</p>
<h2 id="1，质控"><a href="#1，质控" class="headerlink" title="1，质控"></a>1，质控</h2><p> vcf=Variant Call Format 是用于描述SNP，InDel和SV结果的文本文件质控</p>
<p>#查看vcf格式压缩文件：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">zmore 505_maf0.05.int0.9.vcf.gz</span><br></pre></td></tr></table></figure>

<p>染色体设置要比基因组多，不可以少，选择推荐的41：–chr-set 41</p>
<p>#Convert vcf to Plink format：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --vcf 505_maf0.05.int0.9.vcf.gz --chr-set 41 --make-bed --out 505_maf0.05.int0.9 </span><br></pre></td></tr></table></figure>

<p>#为缺少rs标识符的SNP分配唯一的标识符：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_maf0.05.int0.9 --set-missing-var-ids @:#[c41]\$1,\$2 --chr-set 41 --make-bed --out 505_maf0.05.int0.9_no_missing_IDs</span><br></pre></td></tr></table></figure>

<p> [c41]为自己取的名字 </p>
<p>–geno：排除大部分受试者中缺失的 SNP;</p>
<p>-‐mind：排除基因型缺失率高的个体。</p>
<p>建议首先基于宽松的阈值（0.2;&gt; 20％）过滤 SNP 和个体，过滤掉丢失程度很高的 SNP 和个体。然后再应用更严格的过滤参数（0.02）。**注意，应先执行 SNP 过滤再进行个体水平过滤。 </p>
<p> #根据缺失的基因型数据删除SNP：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_maf0.05.int0.9_no_missing_IDs --geno 0.2 --allow-no-sex --chr-set 41 --make-bed --out 505_MDS </span><br></pre></td></tr></table></figure>

<p>#根据缺失的基因型数据删除个体：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS --mind 0.2 --allow-no-sex --chr-set 41 --make-bed --out 505_MDS2</span><br></pre></td></tr></table></figure>

<p> #根据缺失的基因型数据删除SNP：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS2 --geno 0.02 --allow-no-sex --chr-set 41 --make-bed --out 505_MDS3 </span><br></pre></td></tr></table></figure>

<p>#根据缺失的基因型数据删除个体：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS3 --mind 0.02 --allow-no-sex --chr-set 41 --make-bed --out 505_MDS4</span><br></pre></td></tr></table></figure>

<p> ‐‐maf：仅输出高于 MAF 阈值的 SNP。MAF 低的 SNP 很罕见，因此缺乏检测 SNP 表型关联的 power 。这些 SNP 也更容易出现基因分型错误。MAF 阈值应取决于样本量，较大的样本可以使用较低的 MAF 阈值。对于大样本（N = 100,000）与中等样本（N = 10000），通常将 0.01 和 0.05 用作 MAF 阈值。 </p>
<p>#根据最小等位基因频率（MAF）去筛选：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS4 --maf 0.05 --allow-no-sex --chr-set 41 --make-bed --out 505_MDS5</span><br></pre></td></tr></table></figure>

<h2 id="2，群体结构分析"><a href="#2，群体结构分析" class="headerlink" title="2，群体结构分析"></a>2，群体结构分析</h2><p> #将质控之后的505_MDS5文件复制到Structure文件夹进行下一步分析</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cp 505_MDS5.*  /vol3/agis/chengshifeng_group/wuyubing/brassica/guo/02_structure</span><br></pre></td></tr></table></figure>

<p> #LD筛选：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS5 --indep-pairwise 50 10 0.2 --chr-set 41</span><br></pre></td></tr></table></figure>

<p>#将其转化为bed文件：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS5 --extract plink.prune.in --chr-set 41 --make-bed --out 505LD</span><br></pre></td></tr></table></figure>

<p> #安装admixture </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install admixture</span><br></pre></td></tr></table></figure>

<p>#计算K的CV值</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> K <span class="keyword">in</span> 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15;<span class="keyword">do</span>;admixture --cv 505LD.bed <span class="variable">$K</span> | tee <span class="built_in">log</span><span class="variable">$&#123;K&#125;</span>.out;<span class="keyword">done</span>;</span><br></pre></td></tr></table></figure>

<p>#使用grep查看<em>out文件的cv error(交叉验证的误差)值：</em></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">grep -h CV *.out</span><br></pre></td></tr></table></figure>

<p>CV error (K=10): 0.84891</p>
<p>CV error (K=11): 0.84703</p>
<p>CV error (K=12): 0.84678</p>
<p>CV error (K=13): 0.84851</p>
<p>CV error (K=14): 0.84789</p>
<p>CV error (K=15): 0.85001</p>
<p>CV error (K=1): 0.94150</p>
<p>CV error (K=2): 0.90103</p>
<p>CV error (K=3): 0.88261</p>
<p>CV error (K=4): 0.87245</p>
<p>CV error (K=5): 0.86499</p>
<p>CV error (K=6): 0.85874</p>
<p>CV error (K=7): 0.85451</p>
<p>CV error (K=8): 0.85246</p>
<p>CV error (K=9): 0.84955 </p>
<p>有待完善</p>
<p>#使用R作图：</p>
<p>#启用R：</p>
<p>R</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">two=read.table(<span class="string">&quot;505LD.2.Q&quot;</span>)</span><br><span class="line"></span><br><span class="line">head(two)</span><br><span class="line"></span><br><span class="line">pdf(<span class="string">&quot;K2.pdf&quot;</span>,height=<span class="number">10</span>,width=<span class="number">16</span>)</span><br><span class="line"></span><br><span class="line">barplot(t(as.matrix(two)), col=rainbow(<span class="number">2</span>), xlab=<span class="string">&quot;Individual&quot;</span>, ylab=<span class="string">&quot;Ancestry&quot;</span>, border=<span class="literal">NA</span>)dev.off()png(<span class="string">&quot;K2.png&quot;</span>,height=<span class="number">1000</span>,width=<span class="number">1600</span>)barplot(t(as.matrix(two)), col=rainbow(<span class="number">2</span>), xlab=<span class="string">&quot;Individual&quot;</span>, ylab=<span class="string">&quot;Ancestry&quot;</span>, border=<span class="literal">NA</span>)</span><br><span class="line"></span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>

<h2 id="3，亲缘关系分析"><a href="#3，亲缘关系分析" class="headerlink" title="3，亲缘关系分析"></a>3，亲缘关系分析</h2><p>#将QC及筛选后结果复制到kinship文件夹：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cp 505_MDS5.* /vol3/agis/chengshifeng_group/wuyubing/brassica/guo/03_kinship</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --bfile 505_MDS5 --recode 12 --output-missing-genotype 0 --transpose --chr-set 41 -out 505</span><br></pre></td></tr></table></figure>

<p> 生成505.tfam/tped文件</p>
<p>调用emmax:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">emmax;emmax-kin -v -h -d 10 505</span><br></pre></td></tr></table></figure>

<p> 生成（505.hBN.kinf） </p>
<h2 id="4，将表型数据和基因数据整理成一致"><a href="#4，将表型数据和基因数据整理成一致" class="headerlink" title="4，将表型数据和基因数据整理成一致"></a>4，将表型数据和基因数据整理成一致</h2><p>表型数据格式：两列和基因型对应的编号，一列表型数据 </p>
<p>表型最优无偏预测</p>
<h2 id="5，跑gwas"><a href="#5，跑gwas" class="headerlink" title="5，跑gwas"></a>5，跑gwas</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">emmax -v -d 10 -t 505 -o df_gwas_result -p 2phe_gtdianfeng.txt -k 505.BN.kinf -c cov.emmax.txt</span><br></pre></td></tr></table></figure>

<p>df_gwas_result ：输出结果文件名</p>
<p>2phe_gtdianfeng.txt：输入表型文件</p>
<p>505.BN.kinf：输入上一步亲缘关系产生的.kinf文件</p>
<p>cov.emmax.txt：输入协变量文件 </p>
<h2 id="6，结果可视化（曼哈顿图和QQ图）"><a href="#6，结果可视化（曼哈顿图和QQ图）" class="headerlink" title="6，结果可视化（曼哈顿图和QQ图）"></a>6，结果可视化（曼哈顿图和QQ图）</h2><p>在服务器里提交有R脚本的作业 </p>
<p>先写个R脚本后缀 .r,</p>
<p>然后在 .sh脚本里 Rscript  .r 运行R脚本，</p>
<p>然后正常提交作业 </p>
<p>画SNP密度图安装r包CMplot:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">install.packages(<span class="string">&quot;CMplot&quot;</span>)</span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#SNP-density plot</span></span><br><span class="line"></span><br><span class="line">library(CMplot)</span><br><span class="line"></span><br><span class="line">datetwo=read.csv(<span class="string">&quot;two_gwas_result_delete_random.txt&quot;</span>,header=<span class="built_in">T</span>,sep=<span class="string">&quot;\t&quot;</span>)</span><br><span class="line"></span><br><span class="line">head(datetwo)</span><br><span class="line"></span><br><span class="line">CMplot(datetwo,type=<span class="string">&quot;p&quot;</span>,plot.type=<span class="string">&quot;d&quot;</span>,bin.size=<span class="number">1e6</span>,chr.den.col=<span class="built_in">c</span>(<span class="string">&quot;darkgreen&quot;</span>,<span class="string">&quot;yellow&quot;</span>,<span class="string">&quot;red&quot;</span>),file=<span class="string">&quot;jpg&quot;</span>,memo=<span class="string">&quot;&quot;</span>,dpi=<span class="number">600</span>,file.output=<span class="literal">TRUE</span>,verbose=<span class="literal">TRUE</span>,width=<span class="number">9</span>,height=<span class="number">6</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#绘制单性状曼哈顿图</span></span><br><span class="line"></span><br><span class="line">CMplot(datetwo,plot.type=<span class="string">&quot;m&quot;</span>, LOG10=<span class="literal">TRUE</span>, ylim=<span class="literal">NULL</span>, threshold=<span class="built_in">c</span>(<span class="number">1e-3</span>,<span class="number">1e-2</span>),threshold.col=<span class="built_in">c</span>(<span class="string">&#x27;grey&#x27;</span>,<span class="string">&#x27;black&#x27;</span>), threshold.lty = <span class="built_in">c</span>(<span class="number">1</span>,<span class="number">2</span>),threshold.lwd = <span class="built_in">c</span>(<span class="number">1</span>,<span class="number">1</span>), amplify = <span class="built_in">T</span>,signal.cex = <span class="built_in">c</span>(<span class="number">1.5</span>,<span class="number">1.5</span>), signal.pch = <span class="built_in">c</span>(<span class="number">19</span>,<span class="number">19</span>),signal.col = <span class="built_in">c</span>(<span class="string">&quot;firebrick1&quot;</span>))</span><br></pre></td></tr></table></figure>


                
                <hr>
                <div class="row">
                    <div class="col-md-12 post-statement">
                        <span>Author: John Doe</span>
                        <span>Post Link: <a href="http://example.com/2021/07/17/SNPGWASwyb/index.html">http://example.com/2021/07/17/SNPGWASwyb/index.html </a></span>
                        <span>Indicate: 《John Doe》reprint http://example.com/2021/07/17/SNPGWASwyb/index.html，original author: John Doe  </span>
                        <span><a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-nd/3.0/deed.zh">License: Attribution-NonCommercial-NoDerivs 3.0 Unported( CC BY-NC-ND 3.0 )</a></span>
                    </div>
                    <div class="col-md-12">
                        <!-- Categories Start -->
                        <section>
                            <h5><a href="/categories/">CATEGORIES </a></h5>
                            <div class="tags">
                                
                            </div>
                        </section>
                        <!-- Categories End -->
                    </div>
                    <div class="col-md-12">
                        <!-- Tags Start -->
                        <section class="past-tags">
                            <h5><a href="/tags/">TAGS </a></h5>
                            <div class="tags">
                                
                            </div>
                        </section>
                        <!-- Tags End -->
                    </div>
                </div>

                <hr>
                <!-- Pager Start -->
                <ul class="pager">
                    
                    
                        <li class="next">
                            <a href="/2021/07/17/hello-world/" data-toggle="tooltip" data-placement="top" title="Hello World">NEXT POST  &rarr;</a>
                        </li>
                    
                </ul>
                <!-- Pager End -->
               
                <!-- Comment Start -->
<div class="comment">
  
  

  

  
</div>            
<!-- Comment End -->
                
                <br>
                <!-- Comment Start -->
<div class="comment">
  
  

  

  
</div>            
<!-- Comment End -->
            </div>
            
            <!-- Tabe of Content -->
            

<script>
let toc = document.getElementById('toc')
if (toc != null) {
    window.addEventListener("scroll", scrollcatelogHandler);
    let tocPosition = toc.offsetTop;
    function scrollcatelogHandler(e) {
        let event = e || window.event,
            target = event.target || event.srcElement;
        let scrollTop = document.documentElement.scrollTop || document.body.scrollTop;
        if (scrollTop > tocPosition - 60) {
            toc.classList.add("toc-fixed");
        } else {
            toc.classList.remove("toc-fixed");
        }
    }
}
</script>
        </div>
    </div>
</article>

    </div>
    <!-- Main Content End -->

    <span id="rocket"></span>

    <!-- Google Analytics -->
    

    <!-- Baidu Tongji -->
    

    <script>
        // TOC scroll
        $(".toc-nav-link").bind("click touch", function () {
            // 根据a标签的href转换为id选择器，获取id元素所处的位置
            $('html,body').animate({ scrollTop: ($($(this).attr('href')).offset().top) }, 300);
        });
    </script>
</body>
</html>
